share/functions/fish_fossil_prompt.fish

```diff
@@ -4,48 +4,88 @@ function fish_fossil_prompt --description 'Write out the fossil prompt'
         return 1
     end
 
-    # Bail if not a fossil checkout
-    if not fossil ls &> /dev/null
-	return 127
-    end
+    # Read branch and bookmark (bail if not checkout)
+    set -l branch (fossil branch current 2>/dev/null)
+    or return 127
+
+    set -g fish_color_fossil_clean green
+    set -g fish_color_fossil_modified yellow
+    set -g fish_color_fossil_dirty red
+
+    set -g fish_color_fossil_added green
+    set -g fish_color_fossil_renamed magenta
+    set -g fish_color_fossil_missing red
+    set -g fish_color_fossil_deleted red
+    set -g fish_color_fossil_untracked yellow
+    set -g fish_color_fossil_conflict red
+
+    set -g fish_prompt_fossil_status_added '✚'
+    set -g fish_prompt_fossil_status_modified '*'
+    set -g fish_prompt_fossil_status_renamed '⇒'
+    set -g fish_prompt_fossil_status_deleted '-'
+    set -g fish_prompt_fossil_status_missing '✖'
+    set -g fish_prompt_fossil_status_untracked '?'
+    set -g fish_prompt_fossil_status_conflict '×'
+
+    set -g fish_prompt_fossil_status_order added modified renamed deleted missing untracked conflict
 
-    # Parse fossil info
-    set -l fossil_info (fossil info)
-    string match --regex --quiet \
-        '^project-name:\s*(?<fossil_project_name>.*)$' $fossil_info
-    string match --regex --quiet \
-        '^tags:\s*(?<fossil_tags>.*)$' $fossil_info
 
     echo -n ' ['
-    set_color --bold magenta
-    echo -n $fossil_project_name
-    set_color normal
-    echo -n ':'
-    set_color --bold yellow
-    echo -n $fossil_tags
-    set_color normal
+    # Disabling color and pager is always a good idea.
```

Comment:
I don't know what this comment refers to, is it outdated?

Reply:
Seems like it. 
https://github.com/fish-shell/fish-shell/blob/master/share/functions/fish_hg_prompt.fish#L47